dsp-user/core/src/main/scala/dsp/user/domain/UserDomain.scala

```diff
@@ -5,138 +5,189 @@
+    new User(
+      self.id,
+      self.givenName,
+      self.familyName,
+      newValue,
+      self.email,
+      self.password,
+      self.language,
+      self.status
+    ) {}
 
-  //   value match {
-  //     case uuidPattern(value) => new UserId(UUID.fromString(value), Iri.UserIri.make(value)) {}
-  //     case iriPattern(value) =>
-  //       new UserId(UUID.fromString(value.substring(value.lastIndexOf("/") + 1)), Iri.UserIri.make(value)) {}
-  //     //case _                  => ???
-  //   }
-  // }
+  /**
+   * Update the email of a user
+   *
+   *  @param id  the user's ID
+   *  @param newValue  the new email
+   *  @return the updated [[User]]
+   */
+  def updateEmail(newValue: Email): User =
+    new User(
+      self.id,
+      self.givenName,
+      self.familyName,
+      self.username,
+      newValue,
+      self.password,
+      self.language,
+      self.status
+    ) {}
 
   /**
-   * Generates a UserId instance with a new (random) UUID and an IRI which is created from a prefix and the UUID.
+   * Update the given name of a user
    *
-   * @return a new UserId instance
+   *  @param id  the user's ID
+   *  @param newValue  the new given name
+   *  @return the updated [[User]]
    */
-  def fromIri(iri: Iri.UserIri): UserId = {
-    val uuid: UUID = UUID.fromString(iri.value.split("/").last)
-    new UserId(uuid, iri) {}
-  }
+  def updateGivenName(newValue: GivenName): User =
+    new User(
+      self.id,
+      newValue,
+      self.familyName,
+      self.username,
+      self.email,
+      self.password,
+      self.language,
+      self.status
+    ) {}
 
   /**
-   * Generates a UserId instance with a new (random) UUID and an IRI which is created from a prefix and the UUID.
+   * Update the family name of a user
    *
-   * @return a new UserId instance
+   *  @param id  the user's ID
+   *  @param newValue  the new family name
+   *  @return the updated [[User]]
    */
-  def fromUuid(uuid: UUID): UserId = {
-    val iri: Iri.UserIri = Iri.UserIri.make("http://rdfh.ch/users/" + uuid.toString)
-    new UserId(uuid, iri) {}
-  }
+  def updateFamilyName(newValue: FamilyName): User =
+    new User(
+      self.id,
+      self.givenName,
+      newValue,
+      self.username,
+      self.email,
+      self.password,
+      self.language,
+      self.status
+    ) {}
 
   /**
-   * Generates a UserId instance with a new (random) UUID and an IRI which is created from a prefix and the UUID.
+   * Update the password of a user
    *
-   * @return a new UserId instance
+   *  @param id  the user's ID
+   *  @param newValue  the new password
+   *  @return the updated [[User]]
    */
-  // TODO should this return a Validation[Throwable, UserId]
-  def make(): UserId = {
-    val uuid: UUID       = UUID.randomUUID()
-    val iri: Iri.UserIri = Iri.UserIri.make("http://rdfh.ch/users/" + uuid.toString)
-    new UserId(uuid, iri) {}
-  }
-}
+  def updatePassword(newPassword: PasswordHash): User =
```

Comment:
To be consistent with line 124, you should use `newValue` here ;)
Also, `id` isn't param anymore in all update methods, is it?.